guarddog/utils/archives.py

```diff
@@ -1,49 +1,34 @@
 import logging
 import os
-import zipfile
 import stat
+import zipfile
+
 import tarsafe  # type:ignore
 
 log = logging.getLogger("guarddog")
 
 
 def is_supported_archive(path: str) -> bool:
     """
-    Decide whether a file contains a supported archive.
+    Decide whether a file contains a supported archive based on its
+    file extension.
 
     Args:
         path (str): The local filesystem path to examine
 
     Returns:
         bool: Represents the decision reached for the file
-    """
-    return is_tar_archive(path) or is_zip_archive(path)
-
 
-def is_tar_archive(path: str) -> bool:
     """
-    Decide whether a file contains a tar archive.
+    def is_tar_archive(path: str) -> bool:
+        tar_exts = [".bz2", ".bzip2", ".gz", ".gzip", ".tgz", ".xz"]
 
-    Args:
-        path (str): The local filesystem path to examine
+        return any(path.endswith(ext) for ext in tar_exts)
 
-    Returns:
-        bool: Represents the decision reached for the file
-    """
-    return any(path.endswith(ext) for ext in [".tar.gz", ".tgz"])
+    def is_zip_archive(path: str) -> bool:
+        return any(path.endswith(ext) for ext in [".zip", ".whl", ".egg"])
 
-
-def is_zip_archive(path: str) -> bool:
-    """
-    Decide whether a file contains a zip, whl or egg archive.
-
-    Args:
-        path (str): The local filesystem path to examine
-
-    Returns:
-        bool: Represents the decision reached for the file
-    """
-    return any(path.endswith(ext) for ext in [".zip", ".whl", ".egg"])
+    return is_tar_archive(path) or is_zip_archive(path)
```

Comment:
is this correct?

Reply:
Yup, you're just checking that we have some file extension for an archive we know how to extract